src/api/index.ts

```diff
@@ -30,6 +46,62 @@ apiRouter.get('/', maybeAuthenticated, (req: Request, res: Response) => {
     .then(topics => res.json(topics));
 });
 
+apiRouter.post('/upload', isAdmin, upload.single('file'),
+  async (req: Request, res: Response) => {
+    const data = req.file.buffer.toString('utf8');
+    const destination = path.join(uploadContentDir, req.file.originalname);
+    try {
+
+      validateDocumentName(req.file);
+      validateDocumentPayload(data);
+
+      await fs.promises.writeFile(destination, data);
+      await syncContent(uploadContentDir);
+      res.sendStatus(200);
+    } catch (error) {
+      const isFSError = handleNodeFSErrors(error);
+      if (isFSError) {
+        return res.status(500).json({ error: 'Oops! something went wrong.' });
+      }
+
+      res.status(400).json({ error: error.message });
+    }
+  });
+
+apiRouter.delete('/remove-topic/:filename', isAdmin,
+  async (req: Request, res: Response) => {
+    try {
+      await removeContentAndTopic(req.params.filename);
```

Comment:
After removing a file/topic we alson need to rebuild the autocomplete index to remove any words added by the file. At present, the simplest way would seem to me to simply delete the file and then call `syncContent` again.

BTW, we should only delete files from the `content` directory, not from the (version-managed) `arab-content` repo.